lib/model.js

```diff
@@ -2719,11 +2722,11 @@ class Model {
         }
       }
 
-      if (options.ignoreDuplicates && ['mssql', 'db2'].includes(dialect)) {
+      if (options.ignoreDuplicates && ['mssql', 'db2', 'ibmi'].includes(dialect)) {
         throw new Error(`${dialect} does not support the ignoreDuplicates option.`);
       }
 
-      if (options.updateOnDuplicate && (dialect !== 'mysql' && dialect !== 'mariadb' && dialect !== 'sqlite' && dialect !== 'postgres')) {
+      if (options.updateOnDuplicate && (dialect !== 'mysql' && dialect !== 'mariadb' && dialect !== 'sqlite' && dialect !== 'postgres' && dialect !== 'ibmi')) {
```

Comment:
I tried to hit these where I could, but I definitely missed a few. Most of the `db2` checks were changed to `.includes` since they went from 1 to 2 . I might skip going through and touching these for now, and then refactor in separate PR